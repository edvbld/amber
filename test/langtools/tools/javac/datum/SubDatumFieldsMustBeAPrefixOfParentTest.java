/*
 * @test /nodynamiccopyright/
 * @summary smoke negative test for datum classes
 * @compile/fail/ref=SubDatumFieldsMustBeAPrefixOfParentTest.out -XDrawDiagnostics SubDatumFieldsMustBeAPrefixOfParentTest.java
 */

public class SubDatumFieldsMustBeAPrefixOfParentTest {
    abstract __datum D1(int x, int y) { }
    __datum D2(int x, int y, int z) extends D1(y, x) { }
}
