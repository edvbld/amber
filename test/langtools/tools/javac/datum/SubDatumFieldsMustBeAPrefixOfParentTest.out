SubDatumFieldsMustBeAPrefixOfParentTest.java:9:44: compiler.err.cant.extend.record: (compiler.misc.super.field.mismatch: int, x, int, y)
1 error
